ncm-spma/src/main/perl/spma/yum.pm

```diff
@@ -907,7 +907,9 @@ sub update_pkgs
         $to_rm = $self->packages_to_remove($wanted);
         defined($to_rm) or return 0;
         $self->spare_dependencies($to_rm, $to_install, $error_is_warn);
-        $tx = $self->schedule(REMOVE, $to_rm);
+        if($#$to_rm > 0){
```

Comment:
I find it hard to keep track of all these special characters, and due to their "special"-ness they are hard to Google. I eventually figured out that $# is "max index", which can be used to approximate length but you have to remember to subtract 1:

```
my $s = [];
my $t = ["x"];
say $#$s;   # -1
say $#$t;   # 0
```

Which means your code will currently skip the pkg removal if $to_rm is a single element list. To be safe, just use scalar:
```
if (scalar(@$to_rm) > 0) {
```

Please could you also follow the local spacing style for ( and { ?

Reply:
Thanks for the tip about array indexes in Perl. My supposedly empty `$to_rm` shows a `$#$to_rm` equal to 0.  This means that is not really empty, complicating this issue.

See the following output in the affected system:
```
$self->debug(3, "List of packages flagged for removal:");
> [DEBUG] List of packages flagged for removal:
$self->debug(3, $to_rm);
> [DEBUG] ()
$self->debug(3, ref($to_rm));
> [DEBUG] Set::Scalar
$self->debug(3, $#$to_rm);
> [DEBUG] 0
$self->debug(3, scalar(@$to_rm));
> [DEBUG] 1
$self->debug(3, length($to_rm->[0]));
> [DEBUG] 0
```

So, `$to_rm` has one element with an empty string...

Reply:
Fixed in https://github.com/quattor/configuration-modules-core/pull/1554/commits/b5376931bbc60621f2e60b98009c7028ce1b9669 and found the reason for the unexpected length of `$to_rm` (see below).